src/polynomials/primitives/bernstein_helpers.cpp

```diff
@@ -0,0 +1,42 @@
+#include "bernstein_helpers.hpp"
+#include "common.hpp"
+
+#include <numeric>
+
+namespace bebot::bernstein::internal
+{
+
+double binom(int n, int k)
+{
+    // Stolen from cppreference :P
+    // TODO: No idea how efficient this is
+    return 1 / ((n + 1) * std::beta(n - k + 1, k + 1));
+}
+
+Eigen::MatrixXd bezier_coefficients(int n)
+{
+    Eigen::MatrixXd coefficients = Eigen::MatrixXd::Zero(n + 1, n + 1);
+    for (auto k : range(0, n + 1)) {
+        for (auto i : range(k, n + 1)) {
+            coefficients(i, k) = pow(-1, i - k) * binom(n, i) * binom(i, k);
```

Comment:
We could put the first `binom` call in the outer loop to minimize the number of times it is called (in my Python implementation binom tended to be a bottleneck in certain situations).